File: evidence/generated-systems/angular/analytics-on-angular/src/app/models/QualityRule.js

// Define collection and schema for QualityRule
export interface QualityRule {
    name:
	type : string
    threshold:
	type : Threshold
    targetField:
	type : string
    Dataset:
	type : Schema.Types.ObjectId
    Checks:
 	type : [{ type: Schema.Types.ObjectId, ref: 'QualityCheck' }]
    Dimension:
 	type : String
    Operator:
 	type : String
#
    collection: 'qualityRules'
}

File: evidence/generated-systems/rubyonrails/analytics-on-rails/analyticsonrails/app/models/qualityrule.rb
class QualityRule < ApplicationRecord
  enum Dimension: [:Completeness, :Accuracy, :Consistency, :Timeliness, :Uniqueness, :Validity]
  enum Operator: [:GreaterThan, :GreaterThanOrEqual, :LessThan, :LessThanOrEqual, :Equal, :NotEqual]


  composed_of :threshold,
    class_name: "Threshold",
    mapping: [
      ${$mapping}, 
      %w[threshold_inclusive inclusive]
    ]

  has_many :Dataset, class_name: 'DataSet'
  has_many :Checks, class_name: 'QualityCheck'

end
